### PR TITLE
fix(config): fix modules not being resolved from correct path

### PR DESCRIPTION
**Pull request by tido64**
### Description

Fix module not being resolved from correct path

### Test plan

CI should pass.

### CODE REVIEW

File: packages/config/package.json [hunk 1]
```diff
@@ -42,7 +42,6 @@
   },
   "devDependencies": {
     "@rnx-kit/eslint-config": "*",
-    "@rnx-kit/jest-preset": "*",
```
Comment (Saadnajmi) [inline packages/config/package.json:45]: duplicate change from https://github.com/microsoft/rnx-kit/pull/3252/files ?

File: packages/config/package.json [hunk 2]
```diff
@@ -42,7 +42,6 @@
...
     "@rnx-kit/metro-plugin-cyclic-dependencies-detector": "*",
```
Comment (Saadnajmi) [inline packages/config/package.json:45]: Sidenote, Should all these be `workspace:*`?

File: packages/config/package.json (hunk 1 above)
Comment (tido64) [reply packages/config/package.json:45]: No, I left this package out:
image

File: packages/config/package.json (hunk 2 above)
Comment (tido64) [reply packages/config/package.json:45]: Actually, I'm not sure whether npm will fail if it encounters `workspace:`. Regardless of whether it's under `devDependencies`. Reverting until we can figure that out.

**Pull request merged**